### ISSUE TITLE
computeHVContr

### ISSUE THREAD

**Issue by mariusbommert** [jakobbossek/ecr2#127]
I used the following code and I am a bit confused why I get a permutation of the hypervolume contributions. The values should have the same order for each computation of the hypervolume contribution. 

```r
library("ecr")
m = matrix(c(1, 8, 2, 4, 3, 0), nrow = 2)
computeHVContr(m, ref.point = c(10, 10))
# 2 4 28
computeHVContr(m, ref.point = c(10, 10))
# 28  2  4
computeHVContr(m, ref.point = c(10, 10))
# 4 28  2
```

**Comment by jakobbossek** [comment]
Hi Marius,

thanks for the report. This should not happen. Internally, `computeHVContr` exchanges columns in the input matrix (see [computeHVContributionC ](https://github.com/jakobbossek/ecr2/blob/master/src/computeHypervolume.c)). Obviously, this is not done on a copy but the original input matrix.
```r
> library(ecr)
> m = matrix(c(1, 8, 2, 4, 3, 0), nrow = 2)
> m
     [,1] [,2] [,3]
[1,]    1    2    3
[2,]    8    4    0
> computeHVContr(m, ref.point = c(10, 10))
[1]  2  4 28
> m
     [,1] [,2] [,3]
[1,]    3    1    2
[2,]    0    8    4
> computeHVContr(m, ref.point = c(10, 10))
[1] 28  2  4
> m
     [,1] [,2] [,3]
[1,]    2    3    1
[2,]    4    0    8
> computeHVContr(m, ref.point = c(10, 10))
[1]  4 28  2
>


```

**Comment by jakobbossek** [comment]
A workaround is to pass a copy.
```r
computeHVContr(m[,], ref.point = c(10, 10))
```
This is done, e.g., in the `selDomHV` selector. Hence, the bug remained undiscovered, but luckily did not affect SMS-EMOA.

**Comment by jakobbossek** [comment]
Fixed via 7a6837f (we now pass a copy down to the c-code).
